pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -186,6 +186,17 @@ func removeResources(elements []common.ClusterObjectReference, index int) []comm
 	return elements[:len(elements)-1]
 }
 
+// getServiceStatus get specified component status
+func (h *AppHandler) getServiceStatus(svc common.ApplicationComponentStatus) common.ApplicationComponentStatus {
```

Comment:
What is this function for?

Reply:
When executing according to the stage, a part of the state information needs to be kept. If the information is completely overwritten, the status will be lost during the execution.